main/src/main/scala/sbt/internal/server/BuildServerProtocol.scala

```diff
@@ -78,10 +78,14 @@ object BuildServerProtocol {
   )
 
   lazy val globalSettings: Seq[Def.Setting[_]] = Seq(
-    bspConfig := BuildServerConnection.writeConnectionFile(
-      sbtVersion.value,
-      (ThisBuild / baseDirectory).value
-    ),
+    bspConfig := {
+      if (bspEnabled.value) {
+        BuildServerConnection.writeConnectionFile(
+          sbtVersion.value,
+          (ThisBuild / baseDirectory).value
+        )
+      } else ()
```

Comment:
`bspConfig` is usually ran intentionally by the user or the client tool. So it is confusing that it does just nothing in the case `bspEnabled := true`. I would at least like to be informed why.

```suggestion
      } else {
         val logger = stream.value.log
         logger.warn("BSP is disabled for this build")
         logger.info("add 'Global / bspEnabled := true' to enable BSP")
      }
```